pkg/template/registry/templateinstance/strategy.go

```diff
@@ -124,20 +130,55 @@ func (s *templateInstanceStrategy) validateImpersonation(templateInstance *templ
 	}
 
 	if templateInstance.Spec.Requester.Username != userinfo.GetName() {
-		sar := authorizationapi.AddUserToSAR(userinfo,
-			&authorizationapi.SubjectAccessReview{
-				Action: authorizationapi.Action{
-					Verb:         "impersonate",
-					Group:        userapi.GroupName,
-					Resource:     authorizationapi.UserResource,
-					ResourceName: templateInstance.Spec.Requester.Username,
-				},
-			})
-		resp, err := s.oc.SubjectAccessReviews().Create(sar)
-		if err != nil || resp == nil || !resp.Allowed {
+		if err := util.Authorize(s.kc.Authorization().SubjectAccessReviews(), userinfo, &authorization.ResourceAttributes{
+			Namespace: templateInstance.Namespace,
+			Verb:      "assign",
+			Group:     templateapi.GroupName,
+			Resource:  "templateinstances",
+		}); err != nil {
 			return field.ErrorList{field.Forbidden(field.NewPath("spec.impersonateUser"), "impersonation forbidden")}
 		}
 	}
 
 	return nil
 }
+
+type statusStrategy struct {
+	runtime.ObjectTyper
+	names.NameGenerator
+}
+
+var StatusStrategy = statusStrategy{kapi.Scheme, names.SimpleNameGenerator}
+
+func (statusStrategy) NamespaceScoped() bool {
+	return true
+}
+
+func (statusStrategy) AllowCreateOnUpdate() bool {
+	return false
+}
+
+func (statusStrategy) AllowUnconditionalUpdate() bool {
+	return false
+}
+
+func (statusStrategy) PrepareForCreate(ctx apirequest.Context, obj runtime.Object) {
```

Comment:
you shouldn't have these in the interface since you only need an `UpdateStrategy`